storage/innobase/include/buf0buf.h

```diff
@@ -1728,8 +1728,10 @@ class buf_pool_t
   {
     /** Number of array[] elements per page_hash_latch.
     Must be one less than a power of 2. */
-    static constexpr size_t ELEMENTS_PER_LATCH= CPU_LEVEL1_DCACHE_LINESIZE /
+    static constexpr size_t ELEMENTS_PER_LATCH= 64 /
       sizeof(void*) - 1;
+    static constexpr size_t EMPTY_SLOTS_PER_LATCH =
+      ((CPU_LEVEL1_DCACHE_LINESIZE / 64) - 1) * (64 / sizeof(void*));
```

Comment:
I see that `EMPTY_SLOTS_PER_LATCH` will compute to 0 on systems with 64-byte cache line size.

I think that we want
```c++
static_assert(CPU_LEVEL1_DCACHE_LINESIZE >= 64, "less than 64 bytes");
static_assert(!(CPU_LEVEL1_DCACHE_LINESIZE & 63), "not a multiple of 64 bytes");
```
somewhere in this class to document the underlying assumptions. It would seem that the cache line size would be 64, 128, or 256 bytes.

I see that on s390x the cache line size is 256 bytes. @grooverdan, we might want to run a test there.

Formatting nitpick: In the common server code formatting style that was previously not used by InnoDB, there is not supposed to be a space before the assignment operator: `x= y;`

Reply:
No objection. I don't have enough of a dedicated s390x access, or significant time, to really perform performance tests. I (used to) generally try to fill cache line structures with a required amount of needed data that can be passively fetched in one cache miss than crudely pad it to useless unused data. If there's no tradeoff in that relm ok.

Reply:
static_assert added. formatting fixed.